src/main/java/blue/UI.java

```diff
@@ -0,0 +1,96 @@
+package blue;
+
+import java.util.ArrayList;
+
+import blue.task.Task;
+
+
+
+/**
+ * The {@code UI} class handles interactions with the user by displaying messages and task information.
+ * It provides methods to greet the user, say farewell, and display task-related updates.
+ */
+public class UI {
+
+    /**
+     * Displays a greeting message to the user.
+     */
+    public static void greet() {
+        System.out.println("--------------------------------------------");
+        System.out.println("Hello! I'm Blue! Woof Woof! Yap Yap!");
+        System.out.println("What can I do for you?");
+        System.out.println("--------------------------------------------");
+    }
+
+    /**
+     * Displays a farewell message to the user.
+     */
+    public static void farewell() {
+        System.out.println("Bye Bye! Hope to see you again soon!");
+        System.out.println("_     /)---(\\          /~~~\\");
+        System.out.println("\\\\   (/ . . \\)        /  .. \\");
+        System.out.println(" \\\\__)-\\(*)/         (_,\\  |_)");
+        System.out.println(" \\_       (_         /   \\@/    /^^^\\");
+        System.out.println(" (___/-(____) _     /      \\   / . . \\");
+        System.out.println("              \\\\   /  `    |   V\\ Y /V");
+        System.out.println("               \\\\/  \\   | _\\    / - \\");
+        System.out.println("                \\   /__'|| \\\\_  |    \\");
+        System.out.println("                 \\_____)|_).\\_).||(__V");
+        System.out.println("--------------------------------------------");
+    }
+
+    /**
+     * Displays a message after a task has been marked as done.
+     *
+     * @param task The task that has been marked as done.
+     */
+    public static void displayAfterMark(Task task) {
```

Comment:
Would it be clearer to rename displayAfterMark to displayTaskMarkedAsDone? This would more explicitly describe what the method is doing, aligning better with standard naming conventions that prioritize clarity.